autoemulate/experimental/data/utils.py

```diff
@@ -93,6 +93,26 @@ def _convert_to_tensors(
             f"Unsupported type for dataset ({type(dataset)}). Must be TensorDataset."
         )
 
+    def _convert_to_numpy(
+        self,
+        x: InputLike,
+        y: InputLike | None = None,
+    ) -> tuple[np.ndarray, np.ndarray | None]:
+        """
+        Convert InputLike x, y to tuple of numpy arrays.
+        """
+        if isinstance(x, np.ndarray) and (y is None or isinstance(y, np.ndarray)):
+            return x, y
+
+        result = self._convert_to_tensors(x, y)
+        if isinstance(result, tuple):
+            x, y = result
+            if y.ndim > 1:  # _convert_to_tensors may return 2D y
```

Comment:
@sgreenbury @radka-j I noticed that the reason we need to adjust the dimensionality here is actually because `_convert_to_numpy` includes a call to `_convert_to_tensors` which we included to avoid duplicating code. Do you think this handles it, or would it be better to refactor `_convert_to_numpy` to not use `_convert_to_tensors`?

Reply:
Or alternatively, do we anticipate there being some cases where y *should* be a 2D numpy array?